server/models/Availability.js

```diff
@@ -0,0 +1,97 @@
+const mongoose = require("mongoose");
+
+const availabilitySchema = new mongoose.Schema({
+  profileId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: 'Profile'
+  },
+  schedule: {
+    'monday' : {
+      startTime: {
+        type: Date
+      }, 
+      endTime: {
+        type: Date
+      },
+      active: {
+        type: Boolean,
+        default: false 
+      }
```

Comment:
Would have been nice to see some more SchemaType options as well as logic here: 
- `start/end` are [Date objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) but all we need is the time part of the Date object, would have been nice to see some date manipulation for this (e.g. set `year/month/day/ms` to some default values [`pre-save/validate`](https://mongoosejs.com/docs/middleware.html#pre) so we can compare them easily),
- Would have been nice to validate `start < end`, 
- Require `start/end` to be set if `active` is set ([example](https://mongoosejs.com/docs/api.html#schematype_SchemaType-required)).

Reply:
What was your expectation regarding the date manipulation? Should i be running some presave function that sanitizes the date to a time, or should I simplify this and just have the start and end times be numbers, so that ex. 12am = 0, 6am = 360, 12pm = 1440

Reply:
@natey37 We can have default values for start/end and also use a `pre-save` function to check whether the start date is less than the end date. In addition to that, if a day is set to `active` - we can make the start/end fields required. Please let me know if you have additional questions